### PR TITLE
[HttpFoundation] AbstractCookieHandlerTest with Response native sendHeaders

### PR DESCRIPTION
**Pull request by cvilleger**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #25344
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc [hunk 1]
```diff
@@ -0,0 +1,151 @@
+<?php
+
+use Symfony\Component\HttpFoundation\Session\Storage\Handler\AbstractSessionHandler;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc]: should be removed

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    $parent = dirname($parent);
+}
+
+require $parent.'/vendor/autoload.php';
+
+error_reporting(-1);
+ini_set('html_errors', 0);
+ini_set('display_errors', 1);
+ini_set('session.gc_probability', 0);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc]: all ini_set for session.* should be removed

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+register_shutdown_function(function () {
+    echo "\n";
+    session_write_close();
+    print_r(headers_list());
+    echo "shutdown\n";
+});
+ob_start();
+
+class TestSessionHandler extends AbstractSessionHandler
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc]: should be removed

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/send_headers_cookie_samesite.php [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Response;
+
+require __DIR__ . '/common.inc';
+
+$r = new Response();
+$r->headers->set('Date', 'Thu, 02 Feb 2012 12:12:12 GMT');
+$r->headers->setCookie(new Cookie('SameSite', 'Strict'));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/Fixtures/send_headers_cookie_samesite.php]: this creates a cookie with name SameSite, and value Strict, that's not the feature we want to test
instead, we want to enable the $sameSite parameter of the constructor of the Cookie class.

File: src/Symfony/Component/HttpFoundation/Tests/HeaderTest.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\Tests;
+
+use PHPUnit\Framework\TestCase;
+
+class HeaderTest extends TestCase
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/HeaderTest.php]: I'd suggest to name this RequestFunctionalTest instead

File: src/Symfony/Component/HttpFoundation/Tests/HeaderTest.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            proc_close(self::$server);
+        }
+    }
+
+    public function testHeader()
+    {
+        $result = file_get_contents('http://localhost:8054/send_headers_cookie_samesite.php');
+        $this->assertStringEqualsFile(__DIR__ . '/Fixtures/send_headers_cookie_samesite.expected', $result);
+    }
+
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/HeaderTest.php]: blank line

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/send_headers_cookie_samesite.php [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Response;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/Fixtures/send_headers_cookie_samesite.php]: not used

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc (hunk 1 above)
Comment (cvilleger) [reply src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc]: Done, thanks.

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc (hunk 2 above)
Comment (cvilleger) [reply src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc]: Done, thanks.

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc (hunk 3 above)
Comment (cvilleger) [reply src/Symfony/Component/HttpFoundation/Tests/Fixtures/common.inc]: Done, thanks.

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/send_headers_cookie_samesite.php (hunk 4 above)
Comment (cvilleger) [reply src/Symfony/Component/HttpFoundation/Tests/Fixtures/send_headers_cookie_samesite.php]: Done, thanks.

File: src/Symfony/Component/HttpFoundation/Tests/HeaderTest.php (hunk 5 above)
Comment (cvilleger) [reply src/Symfony/Component/HttpFoundation/Tests/HeaderTest.php]: Done, thanks.

File: src/Symfony/Component/HttpFoundation/Tests/HeaderTest.php (hunk 6 above)
Comment (cvilleger) [reply src/Symfony/Component/HttpFoundation/Tests/HeaderTest.php]: Removed, thanks.

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/send_headers_cookie_samesite.php (hunk 7 above)
Comment (cvilleger) [reply src/Symfony/Component/HttpFoundation/Tests/Fixtures/send_headers_cookie_samesite.php]: This is my $r below

**Pull request merged**